src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/build.py

```diff
@@ -263,13 +265,25 @@ def acr_queue(cmd,
         result = LongRunningOperation(cmd.cli_ctx)(client_registries.queue_build(
             build_request=build_request, resource_group_name=resource_group_name, registry_name=registry_name))
 
-        print("Queued a build with build-id: {}.".format(result.build_id))
-
         if no_logs == False:
-            print("Starting to stream the logs...")
+            size = os.path.getsize(FILE)
+            unit = ""
+            for S in ['Bytes', 'KB', 'MB']:
+                if size < 1024:
+                    unit = S
+                    break
+                size = size / 1024.0
+            if unit == "" or (unit == "MB" and size > 500):
+                raise Exception("Build context too large. Request declined.")
```

Comment:
`CLIError` instead of just exception

Reply:
There is an except statement at the end of the function before the finally statement namely,  
except Exception as err:
         raise CLIError(err)
that raise CLIerrors altogether

Reply:
I am thinking of keeping consistency. Not sure if it a common practice though

Reply:
A few comments here:
1. Raise in a `try` and catch in its own `except` is an over-complicated flow. Raise `CLIError` is a better pattern IMO.
2. We should narrow the type of the `Exception` to avoid broad-except violation.
3. I wonder if the try-except flow is needed here. Looking at the code in `try` block, `LongRunningOperation` handles errors from `queue_build` for you. Besides the exception you are raising, the only place that can raise exception is from the last return method `acr_build_show_logs`. Can we handle errors there instead?
4. As commented in a different place, nother seems to be printed if `no_logs` is True.